WolvenKit/Views/Tools/ProjectExplorerView.xaml.cs

```diff
@@ -568,49 +570,132 @@ private void RowDragDropController_Dropped(object sender, TreeGridRowDroppedEven
 
         }
 
-        private async Task ProcessFileAction(List<string> sourceFiles, string targetDirectory, bool onlyCopy)
+        /// <summary>
+        ///  Since the previous implementation would sometimes fail silently and claim that perfectly viable files weren't found,
+        /// here's an attempt at implementing everything in a more robust way that's also more in line with windows move/copy behaviour.
+        /// </summary>
+        private async Task ProcessFileAction(IReadOnlyList<string> sourceFiles, string targetDirectory)
         {
-            var controlKeyDown = ViewModel.ModifierViewStateService.GetModifierState(ModifierKeys.Control);
-            
-            foreach (var sourceFile in sourceFiles)
+            var isCopy = ViewModel?.ModifierViewStateService.GetModifierState(ModifierKeys.Control) == true;
+
+            // Abort if a directory is dragged on itself or its parent
+            if (!isCopy && sourceFiles.Count == 1 &&
```

Comment:
Somethings odd here, why does it check for !isCopy twice? Or is that a typo?

Reply:
That was from "CopyOnly" being passed as parameter, good catch!